Readme.md

```diff
@@ -136,6 +140,23 @@ To use winston's existing transports, set `transports` to the values (as in key-
 
 Alternatively, if you're using a winston logger instance elsewhere and have already set up levels and transports, pass the instance into expressWinston with the `winstonInstance` option. The `transports` option is then ignored.
 
+#### `metaField` option
+
+In versions of `express-winston` prior to 4.0.0, this field functioned differently.  
+
+Previously the log entry would always have a "meta" field which would be set to the metadata of the request/error.  
+If `metaField` was set, this information would be stored as an object with the given property on the "meta" object of 
+the log entry.  This prevented the use case where the metadata should be located at the root of the log entry.
+
+In this version, `metaField` defaults to "meta" which maintains the prior versions behavior of storing the metadata at 
+a "meta" property of the log entry.  
+
+Explicitly setting the `metaField` to `null` or "null" causes the metadata to be stored at the root of the log entry.
+
+The `metaField` option now also supports dot separated and array values to store the metadata at a nested location in the log entry.
+
+<h3>Upgrade Note: For those upgrading from a version of `express-winston` prior to 4.0.0 that use the `metaField` property, to keep the same behavior, prepend `meta.` to your current `metaField` configuration. (i.e. 'foo' would become 'meta.foo')</h3> 
```

Comment:
is the breaking change absolutely necessary? I would have thought that we can add a warning to the Readme file with specific instructions for StackDriver.

Reply:
Unfortunately not though I think the "breaking change" is pretty minimal. The original way this library was implemented doesn't follow the description of how the winston library works.  There is no "meta" field on a log entry (that was just an abstraction added by the express-winston library creator).  The Winston library type definition specifically mentions that "metadata" is to be passed as an additional argument to the log message or as fields on the log entry, not as a "meta" property on the log entry.

I at least limited the "breaking change" to a modification of configuration by allowing dots in the `metaField` property (a properly that was almost never used because it caused additional nesting of log messages most people didn't want anyway).  The vast majority of users will not have set this property at all (and my change continues the same behavior should that be the case). Previously setting the property to "null" was identical to not setting the property (or setting it to undefined [which even with this PR still maintains the existing behavior]) so it's unlikely anyone was setting it to that value.  Only if you actually specified the property as a value does the "breaking change" affect you and then only by having to add "meta." before the current configuration to get the same results.